### DISCUSSION TITLE
User-events => filtering 2 event types produces error

### DISCUSSION THREAD

**Discussion by bellerstefan** [keycloak/keycloak#19730] [Q&A] [answered]
Can someone reproduce this Bug?

The following filter on user events

<img width="350" alt="image" src="https://user-images.githubusercontent.com/16673073/232068731-60307388-2286-42ae-b5a0-ab84be2e5aa2.png">

produces this error:

Uncaught server error: java.lang.IllegalArgumentException: No enum constant org.keycloak.events.EventType.LOGIN,REGISTER

KC Version is 21.0.2

**Comment by belfhi** [comment]
facing the same issue, still appears in KC 22.0.1
Any ideas for a workaround?

**Comment by justin-tay** [comment, marked as answer]
You seem to be right this is either a bug in the admin console or the rest endpoint.

https://github.com/keycloak/keycloak/blob/f767eb109e4d419a08be2bb09c42e43b706ee7b2/services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java#L761

The parameter is defined as `@QueryParam("type") List<String> types`

This means that the endpoint expects a request like

`http://keycloak/auth/admin/realms/master/events?first=0&max=11&type=LOGIN&type=REGISTER`

But the admin console is using

`http://keycloak/auth/admin/realms/master/events?first=0&max=11&type=LOGIN%2CREGISTER`

Which requires the backend to be `@QueryParam("type") String types` and then split by the comma.

**Comment by bellerstefan** [comment]
Could be connected/the same issue as #20135

**Reply by ssilvert** [reply]
#20135 should be fixed soon.
